### PR TITLE
Addon Test: Add `@vitest/coverage-v8` in postinstall if no coverage reporter installed

### LINKED ISSUE #29797: Test addon should install coverage provider out of the box
Currently when you install the test addon, it does not install a coverage provider. When you try to run coverage it fails. To debug you need to look at the console, which is non-intuitive. Then you need to add a coverage provider (it suggests V8) and then restart.

We should just install the coverage provider out of the box. And if we think istanbul is the better choice, we should install that and configure Vitest accordingly. (I'm still on the fence about v8 vs instanbul!).

### PR DESCRIPTION
**Pull request by ghengeveld**
Closes #29797 

## What I did

Checks for the presence of `@vitest/coverage-istanbul` or `@vitest/coverage-v8` and when neither is present, installs `@vitest/coverage-v8` as part of the addon's postinstall hook.

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createSize |  0 B | 0 B | 0 B | - | - |
| generateSize |  77.7 MB | 77.7 MB | 1.48 kB | -0.3 | 0% |
| initSize |  133 MB | 133 MB | 8.66 kB | **2.98** | 0% |
| diffSize |  55.1 MB | 55.1 MB | 7.18 kB | **2.96** | 0% |
| buildSize |  6.87 MB | 6.87 MB | 3.78 kB | 1.23 | 0.1% |
| buildSbAddonsSize |  1.51 MB | 1.51 MB | 219 B | **3.33** | 0% |
| buildSbCommonSize |  195 kB | 195 kB | 0 B | - | 0% |
| buildSbManagerSize |  1.86 MB | 1.86 MB | 2.31 kB | **19.19** | 0.1% |
| buildSbPreviewSize |  0 B | 0 B | 0 B | - | - |
| buildStaticSize |  0 B | 0 B | 0 B | - | - |
| buildPrebuildSize |  3.57 MB | 3.57 MB | 2.52 kB | **6.13** | 0.1% |
| buildPreviewSize |  3.3 MB | 3.3 MB | 1.25 kB | 1.15 | 0% |
| testBuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbAddonsSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbCommonSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbManagerSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbPreviewSize |  0 B | 0 B | 0 B | - | - |
| testBuildStaticSize |  0 B | 0 B | 0 B | - | - |
| testBuildPrebuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildPreviewSize |  0 B | 0 B | 0 B | - | - |

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createTime |  7.4s | 7.4s | 28ms | -0.52 | 0.4% |
| generateTime |  21.5s | 21s | -469ms | 0.27 | -2.2% |
| initTime |  13.5s | 13.2s | -306ms | -0.38 | -2.3% |
| buildTime |  8.3s | 8.6s | 282ms | -0.34 | 3.3% |
| testBuildTime |  0ms | 0ms | 0ms | - | - |
| devPreviewResponsive |  5.7s | 4.1s | -1s -556ms | **-1.53** | 🔰-37.2% |
| devManagerResponsive |  4.2s | 3.1s | -1s -114ms | **-1.51** | 🔰-35.4% |
| devManagerHeaderVisible |  576ms | 475ms | -101ms | **-1.44** | 🔰-21.3% |
| devManagerIndexVisible |  677ms | 536ms | -141ms | -1.21 | -26.3% |
| devStoryVisibleUncached |  1.9s | 1.7s | -203ms | -0.03 | -11.6% |
| devStoryVisible |  673ms | 536ms | -137ms | -1.16 | -25.6% |
| devAutodocsVisible |  686ms | 447ms | -239ms | -1.08 | -53.5% |
| devMDXVisible |  562ms | 420ms | -142ms | **-1.27** | 🔰-33.8% |
| buildManagerHeaderVisible |  748ms | 468ms | -280ms | -1.22 | -59.8% |
| buildManagerIndexVisible |  757ms | 555ms | -202ms | -1.11 | -36.4% |
| buildStoryVisible |  596ms | 435ms | -161ms | -1.19 | -37% |
| buildAutodocsVisible |  426ms | 368ms | -58ms | **-1.37** | 🔰-15.8% |
| buildMDXVisible |  510ms | 354ms | -156ms | **-1.66** | 🔰-44.1% |

## Greptile Summary

Adds automatic installation of `@vitest/coverage-v8` during postinstall if no coverage reporter is detected in the Storybook test addon, improving out-of-the-box test coverage functionality.

- Modified `code/addons/test/src/postinstall.ts` to check for existing coverage reporters (`@vitest/coverage-v8` or `@vitest/coverage-istanbul`)
- Added automatic installation of `@vitest/coverage-v8` if no coverage reporter is found
- Added informative console messaging about coverage reporter installation
- Implemented safe installation process with dependency checks

### CODE REVIEW

File: code/addons/test/src/postinstall.ts [hunk 1]
```diff
@@ -219,6 +219,21 @@ export default async function postInstall(options: PostinstallOptions) {
...
+    printInfo(
+      '🙈 Let me cover this for you',
+      dedent`
+        You don't seem to have a coverage reporter installed. Vitest needs either V8 or Istanbul to generate coverage reports.
+
+        Adding ${picocolors.bold(colors.pink(`@vitest/coverage-v8`))} to enable coverage reporting.
+        If you prefer Istanbul, you can remove ${picocolors.bold(colors.pink(`@vitest/coverage-v8`))} and install ${picocolors.bold(colors.pink(`@vitest/coverage-istanbul`))} instead.
+      `
+    );
+    dependencies.push(`@vitest/coverage-v8`); // Version specifier is added below
```
[the remark this answers is not in the document]
Comment (ghengeveld) [reply code/addons/test/src/postinstall.ts:234]: It is.

File: code/addons/test/src/postinstall.ts [hunk 2]
```diff
@@ -219,6 +219,21 @@ export default async function postInstall(options: PostinstallOptions) {
     }
   }
 
+  const v8Version = await packageManager.getInstalledVersion('@vitest/coverage-v8');
+  const istanbulVersion = await packageManager.getInstalledVersion('@vitest/coverage-istanbul');
```
[the remark this answers is not in the document]
Comment (ghengeveld) [reply code/addons/test/src/postinstall.ts:223]: Yes but never for the same package, so caching is useless.

File: code/addons/test/src/postinstall.ts [hunk 3]
```diff
@@ -219,6 +219,21 @@ export default async function postInstall(options: PostinstallOptions) {
...
+  if (!v8Version && !istanbulVersion) {
+    printInfo(
+      '🙈 Let me cover this for you',
+      dedent`
+        You don't seem to have a coverage reporter installed. Vitest needs either V8 or Istanbul to generate coverage reports.
+
+        Adding ${picocolors.bold(colors.pink(`@vitest/coverage-v8`))} to enable coverage reporting.
+        If you prefer Istanbul, you can remove ${picocolors.bold(colors.pink(`@vitest/coverage-v8`))} and install ${picocolors.bold(colors.pink(`@vitest/coverage-istanbul`))} instead.
```
Comment (JReinhold) [inline code/addons/test/src/postinstall.ts:231]: Users would also need to configure `istanbul` to use it. So I don't think we should mention "If you prefer Istanbul..." but instead have a more generic message that points the user to the Vitest Coverage Provider docs for more information.

https://vitest.dev/guide/coverage.html#coverage-providers

**Review by JReinhold — APPROVED** [review]
Minor wording comment.

**Pull request merged**